src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs

```diff
@@ -226,7 +229,9 @@ protected override bool ShouldAnalyzeForEachStatement(ForEachStatementSyntax for
             var newDeclarationTypeNode = newTree.GetRoot(cancellationToken).GetAnnotatedNodes(annotation).Single();
             var newDeclarationType = newSemanticModel.GetTypeInfo(newDeclarationTypeNode, cancellationToken).Type;
 
-            return SymbolEquivalenceComparer.Instance.Equals(declarationType, newDeclarationType);
+            return declarationHasTupleType
+                ? declarationType.Equals(newDeclarationType)
+                : SymbolEquivalenceComparer.Instance.Equals(declarationType, newDeclarationType);
```

Comment:
The right fix is likely to have other SymbolEquivalenceComparer instances.  Including ones that care about name-equivalence or not. I wouldn't necessarily do these as static readonlys (since this would explode the number that there are).  However, i would have them created and cached on demand when someone requests them.  

So, here, you'd ask for a SymbolEquivComparer that cares about tuple element names, and use that to compare the two.